src/dbnode/client/session.go

```diff
@@ -1390,7 +1392,9 @@ func (s *session) writeAttemptWithRLock(
 	state := s.pools.writeState.Get()
 	state.consistencyLevel = s.state.writeLevel
 	state.shardsLeavingCountTowardsConsistency = s.shardsLeavingCountTowardsConsistency
+	state.ShardsLeavingAndInitializingCountTowardsConsistency = s.ShardsLeavingAndInitializingCountTowardsConsistency
 	state.topoMap = s.state.topoMap
+	state.hostSucessMap = make(map[string]bool)
```

Comment:
`state` is being retrieved from an object pool, i.e. don't just allocate here - put it into the object and clean up references on finishing use.